django/db/models/options.py

```diff
@@ -744,16 +737,38 @@ def get_fields(self, include_parents=True, include_hidden=False):
         - include_hidden:  include fields that have a related_name that
                            starts with a "+"
         """
+        if include_parents is False:
+            include_parents = PROXY_PARENTS
         return self._get_fields(include_parents=include_parents, include_hidden=include_hidden)
 
     def _get_fields(self, forward=True, reverse=True, include_parents=True, include_hidden=False,
-                    export_ordered_set=False):
+                    seen_models=None):
+        """
+        Internal helper fuction to return fields of the model. If forward=True then fields defined
+        on this model are returned. If reverse is True, then relations pointing to this model
+        are returned. The include_hidden fields control returning fields with is_hidden attribute
+        set. The include_parents argument toggles if fields from parent models should be included.
+        It has actually three values, True, False and PROXY_PARENTS. When set to PROXY_PARENTS,
+        the call will return all fields defined for the current model or any of its parents in
+        the parent chain to the model's concrete model.
+        """
+        if include_parents not in (True, False, PROXY_PARENTS):
```

Comment:
The include_parents argument *does* have three possible values, and I don't know how else to represent this than with three-valued argument.

I don't mind having True/False/PROXY_PARENTS as the values. We could also have all arguments as objects. For example include_parents=ALL/PROXY_PARENTS/NONE. I don't mind changing this. Any votes what should be done?